rasa_core_sdk/forms.py

```diff
@@ -95,85 +39,64 @@ def name(self):
 
         raise NotImplementedError
 
-    RANDOMIZE = True
+    RANDOMIZE = False
 
     @staticmethod
-    def required_fields():
-        return []
+    def required_slots():
+        raise NotImplementedError
 
     @staticmethod
     def should_request_slot(tracker, slot_name):
         existing_val = tracker.get_slot(slot_name)
         return existing_val is None
 
-    def get_other_slots(self, tracker):
-        # type: (Tracker) -> List[EventType]
-
-        requested_slot = tracker.get_slot(FORM_SLOT_NAME)
-
-        requested_entity = None
-        for f in self.required_fields():
-            if f.slot_name == requested_slot:
-                requested_entity = getattr(f, 'entity_name', None)
-
-        slot_events = []
-        extracted_entities = {requested_entity}
-
-        for f in self.required_fields():
-            if (isinstance(f, EntityFormField)
-                    and f.slot_name != requested_slot
-                    and f.entity_name not in extracted_entities):
-                slot_events.extend(f.extract(tracker))
-                extracted_entities.add(f.entity_name)
-        return slot_events
+    def validate(self, tracker):
+        # type: (Tracker) -> Dict[Text, Any]
+        """"Validate the user input."""
 
-    def get_requested_slot(self, tracker):
-        # type: (Tracker) -> List[EventType]
+        events = []
+        entities = tracker.latest_message["entities"]
 
-        requested_slot = tracker.get_slot(FORM_SLOT_NAME)
-
-        required = self.required_fields()
-
-        if self.RANDOMIZE:
-            random.shuffle(required)
-
-        if requested_slot is None:
-            return []
+        for e in entities:
+            if e.get("entity") == tracker.slots[REQUESTED_SLOT]:
+                events.append(SlotSet(e['entity'], e['value']))
+        if events:
+            return events
         else:
-            fields = [f
-                      for f in required
-                      if f.slot_name == requested_slot]
+            raise InputValidationError("validation not implemented")
```

Comment:
I am not sure how this error message relates to the fact that there are no events?

Reply:
hahaha yes we already said we'd rename this. But wait did we say we're changing the error? I thought only the event?

Reply:
I thought we create ActionExecutionError, that can be returned by any action

Reply:
i thought we were just changing the `ValidationFailed` event to `ActionExecutionFailed`?

Reply:
but if we make event general, why not rename the error as well to sync it with event?